packages/salesforcedx-vscode-lwc/src/commands/forceLightningLwcPreview.ts

```diff
@@ -317,58 +380,158 @@ async function selectPlatformAndExecute(
         logName,
         commandName
       );
-      return;
+      throw e;
     }
   }
 
   // if there are any devices available, show a pick list.
-  if (options.length > 0) {
-    options.unshift({
-      label: createDeviceLabelText,
-      detail: createDeviceDetailText
+  let selectedItem: vscode.QuickPickItem | undefined;
+  if (items.length > 0) {
+    items.unshift(createNewDeviceItem);
+
+    selectedItem = await vscode.window.showQuickPick(items, {
+      placeHolder: nls.localize(
+        'force_lightning_lwc_preview_select_virtual_device'
+      )
     });
 
-    const selectedItem = await vscode.window.showQuickPick(options, {
-      placeHolder: selectDevicePlaceholderText
-    });
-    targetName = selectedItem && selectedItem.label;
-
-    if (targetName === undefined) {
-      vscode.window.showInformationMessage(commandCancelledMessage);
-      return;
+    if (selectedItem === undefined) {
+      // user cancelled operation
+      return undefined;
+    } else {
+      targetName = selectedItem.label;
```

Comment:
This is incorrect, at least for Android. You need the actual `name` of the device. For example, when I select my Nexus 6 emulator, it passes `Nexus 6 API 29` as the `target` to the preview command, but `avd` is going to be looking for `Nexus_6_API_29`, and so it's always going to think that the device does not exist.

You'll have to look up the original `devices` entry, based on one or more of the selected fields, and grab the actual name of the device.

Reply:
ooooh... very good catch. In my local testing I was using an Android emulator named 'emu2' so user friendly display name and device name were the same so I never caught this.